_test/export.response.test.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * General tests for the findologicxmlexport plugin
+ *
+ * @group plugin_findologicxmlexport
+ * @group plugins
+ */
+
+//======================================================================
+// EXPORT RESPONSE TESTS
+//======================================================================
+
+require_once(__DIR__ . '/../DokuwikiXMLExport.php');
+require_once(__DIR__ . '/../admin.php');
+require_once(__DIR__ . '/../_test/Helper.php');
+
+class export_response_test extends DokuWikiTest
+{
+    public function setUp()
+    {
+        Helper::setUp();
+    }
+
+    /**
+     * @dataProvider parameterProviderForXMLResponse
+     */
+    public function test_parameters_start_count_and_total_are_valid_for_($ids = array(), $expectedCount, $expectedTotal)
+    {
+        // If there are no pages, no pages can be saved.
+        if (!$expectedCount == 0 && !$expectedTotal == 0) {
+            Helper::savePages($ids);
+        }
+
+        $xml = Helper::getXML();
+        $start = implode('', ($xml->xpath('/findologic/items/@start')));
+        $count = implode('', ($xml->xpath('/findologic/items/@count')));
+        $total = implode('', ($xml->xpath('/findologic/items/@total')));
+
+        $expectedStart = 0;
+
+        $this->assertEquals($expectedStart, $start, 'Expected start value should match the requested start value.');
+        $this->assertEquals($expectedCount, $count, 'Expected count value should match the requested count value.');
+        $this->assertEquals($expectedTotal, $total, 'Expected total value should match the amount of total pages.');
+    }
+
+    public function parameterProviderForXMLResponse()
+    {
+        return [
+            // name => [[ids], count, total]
+            'no pages' => [[''], 0, 0], // Empty DokuWiki
+            'one page' => [['home'], 1, 1],
+            'two pages' => [['home1', 'home2'], 2, 2],
+            'ten pages' => [['page_1', 'page_2', 'page_3', 'page_4', 'page_5', 'page_6', 'page_7', 'page_8', 'page_9', 'page_10'], 10, 10]
+        ];
+    }
+
+    public function test_parameters_start_count_and_total_are_valid_when_calling_export_with_count_greater_than_total()
+    {
+        $pageId = array();
+        for ($i = 1; $i <= 9; $i++) {
+            $pageId[$i] = 'demopage0' . $i;
+            Helper::savePages(array($pageId[$i]));
```

Comment:
You can just pass `$pageId` to `savePages` after the loop.

Reply:
Because I changed my Method, there are some other cases like in `\Helper::setUp` where I make a foreach that is no longer needed. Thanks for this!